plugins/org.komodo.core/src/org/komodo/repository/search/ContainsClause.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * See the COPYRIGHT.txt file distributed with this work for information
+ * regarding copyright ownership.  Some portions may be licensed
+ * to Red Hat, Inc. under one or more contributor license agreements.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ */
+package org.komodo.repository.search;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import org.komodo.spi.repository.Repository.KeywordCriteria;
+
+/**
+ * A Contains clause
+ *
+ * CONTAINS(tagged.[acme:tagName], 'foo')
+ *
+ */
+class ContainsClause extends Clause {
+
+    private final String alias;
+
+    private final String property;
+
+    private final Set<String> keywords = new LinkedHashSet<String>();
+
+    private final KeywordCriteria keywordCriteria;
+
+    /**
+     * Constructor
+     * @param alias the alias
+     * @param property the property
+     * @param keywordCriteria settings for the keywords
+     * @param keywords the value
+     */
+    public ContainsClause(String alias, String property, KeywordCriteria keywordCriteria, String... keywords) {
+        this.alias = alias;
+        this.property = property;
+        this.keywordCriteria = keywordCriteria;
+        for (String keyword : keywords)
```

Comment:
None of the clauses have error checking in that regard since they are only created via the API methods of ObjectSearcher. The keywords argument is error handled at that level.